tests/functional/modules/test_zos_copy_func.py

```diff
@@ -468,11 +468,57 @@ def test_copy_dir_to_existing_uss_dir_not_forced(ansible_zos_module):
         hosts.all.file(path=dest_dir, state="absent")
 
 
+@pytest.mark.uss
+def test_copy_inner_folders_and_validate_recursive_encoding(ansible_zos_module):
```

Comment:
Good test, I think it would also be really helpful if you added something similar but copying local files and checking that their contents are correct.

I'd also say that `subdirs` instead of `inner_folders` may be a more standard way of calling what you're testing.

Reply:
I'll change it and add the test case